### DISCUSSION TITLE
Macros Don't Work after Initiating Probe

### DISCUSSION THREAD

**Discussion by TexasBuckaroo** [winder/Universal-G-Code-Sender#2054] [Help]
I've just discovered my Macros don't work after I use my touch probe to set my Z height. I have a Dell laptop running Window 10 Home 64 bit. My UGS is the Nightly Build...

Product Version: Universal Gcode Platform 20221104
Java: 13.0.1; OpenJDK 64-Bit Server VM 13.0.1+9
Runtime: OpenJDK Runtime Environment 13.0.1+9
System: Windows 10 version 10.0 running on amd64; Cp1252; en_US (ugsplatform)
User directory: C:\Users\tandb\AppData\Roaming\ugsplatform
Cache directory: C:\Users\tandb\OneDrive\Desktop\ugsplatform-win\${DEFAULT_USERDIR_ROOT}\var\cache

I was using an older version of UGS (v2.0.9); but my laptop needed a new hard drive. So, I had to reinstall Windows 10 and UGS.

Any thoughts on why my Macros won't work after initiating the touch probe?  Didn't have that issue with V2.0.9.

Thanks!

**Comment by daxliniere** [comment]
What are the macros? What is the controller you're interfacing with? What protocol do you have selected in UGS?

**Reply by TexasBuckaroo** [reply]
I have three Macros I created. I'm using a Arduino UNO controller. I'm assuming I'm using the default protocol - I didn't change it. The Macros all work perfectly, as long as I don't use the probe module first, and the probe module also works perfect. Again, I didn't have this issue with v2.0.9.

Thank you!

**Reply by daxliniere** [reply]
Yep, please post the macros. Also, what messages do you get in the Console when you try to run a macro after probing.
So GRBL was already loaded onto the Arduino when you bought it? It's a good idea to post your startup messages here, too.

**Reply by TexasBuckaroo** [reply]
Great. I'll do that in the morning.

Thank you!

**Reply by TexasBuckaroo** [reply]
No message in the Console. No error message. Nothing. It's like the Macro buttons become undefined. You click on a Macro button and nothing happens.

![UGS Macro Window](https://user-images.githubusercontent.com/118316972/202857066-88c0d3e1-8201-4a81-8365-590421ed3a86.PNG)

Again, the Macros work perfectly, and I've had these same Macros for well over a year. Never had an issue before. It's only now that I'm using a later version of UGS this has become an issue.

Clearly, the issue is with this latest version of UGS.

One thing I am doing differently, with this new version of UGS, is I'm probing in inches, rather than millimeters. Yes, my Jog Control is in inches. I always jog in inches.

Are you able to reproduce this, if you probe in inches?
